include/swift/AST/Module.h

```diff
@@ -159,33 +159,48 @@ class OverlayFile;
 /// location.
 class ModuleSourceFileLocationMap;
 
-/// Package unit used to allow grouping of modules by a package name.
-/// ModuleDecl can set PackageUnit as a parent in its DeclContext
-/// See \c ModuleDecl
+/// A unit that allows grouping of modules by a package name.
+/// It's set as a property in ModuleDecl, instead of as its parent decl context,
+/// since otherwise it will break the existing decl context lookups that assume
+/// ModuleDecl as the top level context. See \c ModuleDecl
 class PackageUnit: public DeclContext {
-
+  /// Identifies this package and used for the equality check
   Identifier PackageName;
-
-  PackageUnit(Identifier name);
+  /// Weakly references ModuleDecl that points to this package.
```

Comment:
Is there something that makes this `weakly referenced`? I think I know what you mean but `weak reference` may be misleading here.

Reply:
Updated doc comments per discussion https://github.com/apple/swift/pull/64373